llvmlite/ir/instructions.py

```diff
@@ -162,7 +162,7 @@ def descr(self, buf):
 class PredictableInstr(Instruction):
 
     def set_weights(self, weights):
-        operands = [MetaDataString(self.module, "branch_weights")]
+        operands = [MetaDataString(self.module, b"branch_weights")]
```

Comment:
> Metadata strings contain such things as function names and paths, neither of which are required to be UTF-8, or human-readable text for that matter.

That's true. However, those (function names, file paths) are precisely considered unicode strings on Python 3, so it makes sense that MetaDataString respects this. Or, at least, MetaDataString could encode them using the ASCII codec, which is safer (people would have to encode non-ASCII text manually, which prevents llvmlite from having to guess).